src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,125 @@
+            resize();
+        }
+        int position = hash % table.length;
+        K currKey;
+        if ((currentNode = table[position]) == null) {
+            table[position] = new Node<>(hash, key, value, null);
+        } else {
+            do {
+                if (currentNode.hash == hash && ((currKey = currentNode.key) == key
+                        || (key != null && key.equals(currKey)))) {
+                    currentNode.value = value;
+                    return;
+                }
+                prev = currentNode;
+            } while ((currentNode = currentNode.next) != null);
+            prev.next = new Node<>(hash, key, value, null);
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        Node<K,V> currentNode;
+        K keyCurrent;
+        if (table != null && table.length > 0
+                && (currentNode = table[hash % table.length]) != null) {
+                do {
+                    if (currentNode.hash == hash && ((keyCurrent = currentNode.key) == key
+                            || (key != null && key.equals(keyCurrent))))
+                        return currentNode.value;
+                } while ((currentNode = currentNode.next) != null);
+            }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static int hash(Object key) {
+        int h;
+        return (key == null) ? 0 : (Math.abs(h = key.hashCode()));
+    }
+
+    private void resize() {
+        if (capacity == 0) {
+            capacity = DEFAULT_INITIAL_CAPACITY;
+            loadFactor = DEFAULT_LOAD_FACTOR;
+            threshold = (int)(capacity * loadFactor);
+        } else if (capacity > 0) {
+            if (capacity >= MAXIMUM_CAPACITY) {
+                threshold = Integer.MAX_VALUE;
+            }
+            else {
+                threshold = threshold << 1;
+                capacity = capacity << 1;
+            }
+        }
+        Node<K, V>[] newTable = (Node<K,V>[])new Node[capacity];
+        if (table != null) {
+            transfer(newTable);
+        } else {
+            table = newTable;
+        }
+    }
+
+    private void transfer(Node<K, V>[] newTable) {
+        int newCap = newTable.length;
+        Node<K, V> currNode;
+        int position;
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                position = node.hash % newCap;
+                currNode = newTable[position];
+                if (newTable[position] == null) {
+                    newTable[position] = new Node<K, V>(node.hash, node.key, node.value, null);
+                } else {
+                    while (currNode.next != null) {
+                        currNode = currNode.next;
+                    }
+                    currNode.next = new Node<K, V>(node.hash, node.key, node.value, null);;
```

Comment:
There is an extra semicolon at the end of this line. Remove the redundant semicolon to avoid potential issues.